test/lint/commit-script-check.sh

```diff
@@ -24,10 +24,16 @@ for commit in $(git rev-list --reverse $1); do
     if git rev-list -n 1 --pretty="%s" $commit | grep -q "^scripted-diff:"; then
         git checkout --quiet $commit^ || exit
         SCRIPT="$(git rev-list --format=%b -n1 $commit | sed '/^-BEGIN VERIFY SCRIPT-$/,/^-END VERIFY SCRIPT-$/{//!b};d')"
+        echo "Script: $SCRIPT"
+        SEDREGEX="^([^#'\"\v]|('[^'\v]*'|\"[^\"\v]*\"))*?(?<![a-z\d])sed(?1)*?-i(?:\.\w+)?[\h,]*(?2)[^#'\"\v]*(?2)"
```

Comment:
This seems a bit hard to read and impossible to maintain without comments if this should ever fail. I'd say that a simple comment like "If you are using sed, make sure to not use bsd sed syntax" on failure is sufficient.

Reply:
So you don't need me to check for sed bsd syntax or you want the regex to be simpler? I can ignore comments and strings and revert to an older commit.

Reply:
Or should I add comments to explain the regex better?

Reply:
I am not sure myself, but I think I'd rather not do the check. It just seems too fragile, especially with the non-greedy regex, which often isn't implemented correctly in regex engines. And how would we check that the `grep --perl-regexp` is using the right regex engine, and not maybe a BSD regex engine? Add another check ...?

Reply:
Ok, how about I just grep for sed without any regular expressions then? Do you want me to undo my commits or to just add a new one?